From bb2912597a37d52e99891510d8c43352127457dd Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Wed, 15 Jan 2025 02:16:29 +0000
Subject: [PATCH] Allow compile with -std=c23

When compiling with -std=c23 the following errors are observed.

    common/qtypes.h:37:15: error: cannot use keyword 'false' as enumeration constant
       37 | typedef enum{ false, true, ENSURE_INT_QBOOL = 0x70000000 } qboolean;
          |               ^~~~~
    common/qtypes.h:37:23: error: cannot use keyword 'true' as enumeration constant
       37 | typedef enum{ false, true, ENSURE_INT_QBOOL = 0x70000000 } qboolean;
          |                       ^~~~
Fix:
- update enum as false0 and true1 as the enum are already 0 and 1, and map to false/true
---
 common/qtypes.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/qtypes.h b/common/qtypes.h
index 97633fc..3ea14a0 100644
--- a/common/qtypes.h
+++ b/common/qtypes.h
@@ -33,7 +33,7 @@ typedef unsigned char byte;
 typedef enum{ false, true };
 typedef int qboolean;
 #else
-typedef enum{ false, true, ENSURE_INT_QBOOL = 0x70000000 } qboolean;
+typedef enum{ false0, true1, ENSURE_INT_QBOOL = 0x70000000 } qboolean;
 #endif
 
 #ifndef offsetof
-- 
2.51.0

